src/main/java/com/auth0/net/Telemetry.java

```diff
@@ -4,50 +4,81 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.commons.codec.binary.Base64;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 @SuppressWarnings("WeakerAccess")
-class Telemetry {
+public class Telemetry {
     static final String HEADER_NAME = "Auth0-Client";
 
     private static final String NAME_KEY = "name";
     private static final String VERSION_KEY = "version";
+    private static final String CORE_KEY = "core";
+    private static final String ENV_KEY = "env";
+    private static final String JAVA_KEY = "java";
 
     private final String name;
     private final String version;
+    private final String core;
+    private final Map<String, String> env;
+    private final String value;
 
-    Telemetry(String name, String version) {
-        this.name = name;
-        this.version = version;
-    }
-
-    public String getName() {
-        return name;
+    public Telemetry(String name, String version) {
+        this(name, version, null);
     }
 
-    public String getVersion() {
-        return version;
-    }
+    public Telemetry(String name, String version, String core) {
+        this.name = name;
+        this.version = version;
+        this.core = core;
 
-    public String getValue() {
-        Map<String, String> values = new HashMap<>();
+        Map<String, Object> values = new HashMap<>();
         if (name != null) {
```

Comment:
Would we ever send telemetry if the name was null? Or the version?

Reply:
The only scenario this would happen is if the user overrides it and sets an explicit null for them. I don't think we should send the telemetry in that case. But then we should change that on the android side as well (PR had the same diff and you approved it) auth0/Auth0.Android#209. Side note, env will always have at least the java version regardless. But again, without name and version it wouldn't make sense. Your call!

Reply:
I don't know what the behavior in the reports would be with a blank name or version. Name is the most critical, without that we have nothing to go off of so I would skip sending in that case. For version ... that would likely just be a mistake and if it was a non-Auth0 developer extending for some reason, they wouldn't see the problem. So ... name required, I don't think version needs to be. Thoughts?

Sorry if I missed that in the other library. If there are questions around how the telemetry should behave, can you call that out with a comment next time so it's not missed?